src/memberdef.cpp

```diff
@@ -4810,24 +4814,52 @@ void MemberDefImpl::enableCallGraph(bool e)
   if (e) Doxygen::parseSourcesNeeded = TRUE;
 }
 
+void MemberDefImpl::enableCallGraph(bool e1, bool e2)
+{
+  if (e1 == e2) m_hasCallGraph=e1;
+  else m_hasCallGraph=!Config_getBool(CALL_GRAPH);
```

Comment:
is the `!` here correct? Feels counterintuitive to assign the negation of `CALL_GRAPH` to `m_hasCallGraph`

Reply:
Yes it is correct, at startup the value of `m_hasCallGraph` (actually through `callGraph` in the `Entry` class) is `Config_getBool(CALL_GRAPH)` as here `e1` and `e2` differ it means that a setting contrary to `Config_getBool(CALL_GRAPH)` is used and thus should here `!Config_getBool(CALL_GRAPH)` be used.

Reply:
Ok, I see the logic now, I think it may be clearer to replace the `enableCallGraph(bool,bool)` with a `mergeEnableCallGraph(bool)`, like so:

```c++
    void mergeEnableCallGraph(bool other)
    {
      if (Config_getBool(CALL_GRAPH))
      {
        enableCallGraph(m_hasCallGraph && other); // enabled if neither deviate from config value
      }
      else
      {
        enableCallGraph(m_hasCallGraph || other); // enabled if either deviate from config value
      }
    }
```

Reply:
Makes sense. When starting the change for the the `||`, the easiest was to change from 1 to 2 arguments and place the logic in the function. When looking at the code it shows that always am other value is combined with the current value and so a "merge" is more appropriate.
Regarding the the logic in the merge function multiple formulations are possible and all of them have their typical drawback in interpreting them.

I've updated the proposed PR